mpi_fortran_handle_types.F90

```diff
@@ -0,0 +1,54 @@
+module mpi_fortran_handle_types
+
+    type, bind(C) :: MPI_Status
```

Comment:
This is funny... Fortran `bind(c)` types with Fortran integers slots :exploding_head:.

Reply:
See MPI 4.1 2.5.1
```fortran
TYPE, BIND(C) :: MPI_Comm
  INTEGER :: MPI_VAL 
END TYPE MPI_Comm
```

Reply:
Oh, I'm aware of that... I was trying to be sarcastic about the MPI standard itself. This is a supposedly C-interoperable type that is not actually interoperable :laughing:.

Reply:
There's even a rationale about it, FWIW.